superset-frontend/src/explore/components/controls/FilterControl/AdhocFilterEditPopoverSimpleTabContent.jsx

```diff
@@ -379,12 +386,25 @@ export default class AdhocFilterEditPopoverSimpleTabContent extends React.Compon
               name="filter-value"
               {...comparatorSelectProps}
               getPopupContainer={triggerNode => triggerNode.parentNode}
+              onSearch={val => this.setState({ currentSuggestionSearch: val })}
+              onSelect={this.clearSuggestionSearch}
+              onBlur={this.clearSuggestionSearch}
             >
               {this.state.suggestions.map(suggestion => (
                 <Select.Option value={suggestion} key={suggestion}>
                   {suggestion}
                 </Select.Option>
               ))}
+
+              {/* enable selecting an option not included in suggestions */}
+              {currentSuggestionSearch &&
+                !this.state.suggestions.some(
+                  suggestion => suggestion === currentSuggestionSearch,
+                ) && (
+                  <Select.Option value={currentSuggestionSearch}>
+                    {currentSuggestionSearch}
+                  </Select.Option>
+                )}
```

Comment:
You probably don't need this since AntD select [has a "tags" mode](https://ant.design/components/select/#components-select-demo-tags), which behave similar to a freeForm react-select (`CreatableSelect`).

Reply:
The problem is that `tags` allows to select multiple values, and operators like `equals` should work with a single value. That's why this "hack" was needed

Reply:
Thanks for the clarification!

Things like this makes me wonder [again](https://github.com/apache/superset/pull/12649) whether we should revert back to react-select... It seems AntD select simply doesn't have the feature completeness and flexibility we want. If this is about aesthetic improvement, we can always restyle react-select to look like AntD (or whatever new style we like).

Reply:
I see your concerns and they are all valid. Since we have gone so far in the migration already, it will be a major time waste reverting back. @kgabryje @villebro @zhaoyongjie let's allocate sometime to investigate and understand limitations of AntD, and have a long-term plan to address those concerns before diving deeper into renovating control panel.

Reply:
AFAIK, the only places we have replaced react-select are in the AdhocFilter and MetricsControl popovers. There are many more places that are currently using react-select. Please also consider future migration cost if we decide to all-in AntD select.

Note that react-select is still a very much actively maintained library, and have some powerful backers.

Reply:
is there a reason why we didn't identify limitations or raise concern before the SIP was voted in?

Reply:
As far as I can tell, from an end user perspective this UX is indistinguishable from something where this would be natively supported without the manually added `Select.Option`. As this solution is quite simple and not really a hack IMO (we are still using fully standard components here), I feel this is fine as-is.

Reply:
Do note though you'd have to do this for everywhere the current `CreatableSelect` is used, ultimately. So this hack might need to be generalized to a wrapper component if we are to keep the same UX.

It just doesn't feel right to me to introduce a hacky solution when there is a current solution that have worked perfectly fine.

Reply:
Not saying we should definitely revert. Just laying out some concerns here.

Reply:
I think adding a wrapper that hides the specifics of how the component actually implements the creatable option sounds like a good option. To my understanding this is essentially how the migration process has been carried out this far, i.e. importing components from `src/common/components/`, where we start out by just exporting the vanilla component, and later extending the component if needed. This way we can later much easier swap out the implementation of said component if a new better library comes along (or if the current implementation ends up being more problematic than we had envisioned).

Btw, I looked at what the current react-select implementation looks like in the Filter Box, and that flow seems quite hacky to the end user (see below). The select control that's used in the control panel isn't as bad, but it does go to show that the react-select experience in Superset has not been optimal from a UX perspective, either.
![filterbox_select](https://user-images.githubusercontent.com/33317356/107879908-4e49f280-6ee4-11eb-93e2-1781fbcf6019.gif)

Reply:
Thank you @villebro, for further explanation. i honestly don't have enough knowledge to tell which library has less drawbacks or better extensibility. I am sure we can always tweak UI/UX and make improvement on either one. since we landed on AntD, let's stick with it and invest in one direction. also, thanks @ktmud  for raising your concerns. Let's leave more time for discussion next time when it comes major changes like this. I read through the SIP and felt that we should have done more research back then......